cmd/internal/cli/pull.go

```diff
@@ -228,13 +232,69 @@ func pullRun(cmd *cobra.Command, args []string) {
 			sylog.Fatalf("While creating Docker credentials: %v", err)
 		}
 
-		libexec.PullOciImage(name, args[i], types.Options{
-			TmpDir:           tmpDir,
-			Force:            force,
-			NoHTTPS:          noHTTPS,
-			DockerAuthConfig: authConf,
-			NoCleanUp:        noCleanUp,
-		})
+		sysCtx := &ocitypes.SystemContext{
```

Comment:
This function is already too long to be adding more lines to it :-|

Can you please move the whole ociclient.IsSupported(transport) case to a separate function, please?